src/Microsoft.Health.Fhir.Core/Models/ResourceChangeData.cs

```diff
@@ -0,0 +1,24 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+
+namespace Microsoft.Health.Fhir.Core.Models
+{
+    public class ResourceChangeData : IResourceChangeData
```

Comment:
>ResourceChangeData [](http://example.com/codeflow?start=17&length=18)

I would make this class as `record`, i.e. immutable with only place where you can set values is constuctor.

It's generally dangerous to keep setters public for something which repressent DTO.

Reply:
You're right. Thank you for pointing it out. At this moment, I would like to keep it as a POCO and add this item using the "record" to our backlog due to there is no usage of "record" in our code base, and as a new feature, it might require a version upgrade for the healthcare shared components.

Reply:
It doesn't have to be `record`, but I think it would be good if this class had immutable properties. I would assume this information is important and to give anyone in the code ability to change it's values sounds a bit dangerous.

Reply:
Made the class immutable.